packages/client/src/graphql/Authentication.re

```diff
@@ -44,20 +40,22 @@ let component =
 
   let query = Query.make(~auth0jwt, ~auth0accessToken, ~authToken, ());
 
-  QueryComponent.make(~variables=query##variables, ({result}) =>
-    result
-    |> HttpResponse.fromApollo
-    |> HttpResponse.fmap(e => e##authentication##jwt)
-    |> (
-      e =>
-        switch (e) {
-        | Success(c) =>
-          ServerJwt.set(c);
-          reload();
-          _innerComponent;
-        | _ => redirectingMessage
-        }
-    )
-  )
-  |> E.React.el;
+  <QueryComponent variables=query##variables>
+    ...{({result}) =>
+      result
+      |> HttpResponse.fromApollo
+      |> HttpResponse.fmap(e => e##authentication##jwt)
+      |> (
+        e =>
+          switch (e) {
+          | Success(c) =>
+            // @todo: rapir this hard logic
```

Comment:
This too. Maybe some refactoring story or similar?

Reply:
Yes, have added new issue: https://github.com/OAGr/foretold/issues/434